src/Mutator/NodeMutationGenerator.php

```diff
@@ -113,22 +130,7 @@ private function generateForMutator(Node $node, Mutator $mutator): iterable
             );
         }
 
-        $isOnFunctionSignature = $node->getAttribute(ReflectionVisitor::IS_ON_FUNCTION_SIGNATURE, false);
-
-        if (!$isOnFunctionSignature
-            && !$node->getAttribute(ReflectionVisitor::IS_INSIDE_FUNCTION_KEY)
-        ) {
-            return;
-        }
-
-        $tests = $this->trace->getAllTestsForMutation(
-            $this->lineRangeCalculator->calculateRange($node),
-            $isOnFunctionSignature
-        );
-
-        if ($tests instanceof Traversable) {
-            $tests = iterator_to_array($tests, false);
-        }
+        $tests = $this->getAllTestsForMutation();
```

Comment:
under what circumstances is this called multiple times?

Reply:
We can have multiple mutations for a give node, can't we?

Reply:
This probably be best renamed to something like `getAllTestsForNode`